ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v3/fileinput/clientlibs/editor/js/editDialog.js

```diff
@@ -0,0 +1,94 @@
+ * limitations under the License.
+ ******************************************************************************/
+(function($) {
+    "use strict";
+    let EDIT_DIALOG = ".cmp-adaptiveform-fileinput__editdialog",
+        FILEINPUT_MIMETYPE = EDIT_DIALOG + " .cmp-adaptiveform-fileinput__mimeType",
+        FILEINPUT_EXTENSIONS = EDIT_DIALOG + " .cmp-adaptiveform-fileinput__extensions",
+        Utils = window.CQ.FormsCoreComponents.Utils.v1;
+
+    function handleMimeType(dialog) {
+        let mimeTypeWrapper = dialog.find(FILEINPUT_MIMETYPE);
+        let mimeTypeField = mimeTypeWrapper.find("coral-multifield");
+        let extensionsField = dialog.find(FILEINPUT_EXTENSIONS + " coral-multifield");
+        let mimeTypeFieldInfoIcon = mimeTypeWrapper.find('coral-icon.coral-Form-fieldinfo');
+        
+        function updateMimeTypeState() {
+            let hasExtensions = extensionsField.find('coral-multifield-item').length > 0;
+            
+            // Disable the entire multifield and its contents
+            mimeTypeField.css({
+                'pointer-events': hasExtensions ? 'none' : 'auto',
+                'opacity': hasExtensions ? '0.5' : '1'
+            });
+            mimeTypeField.prop('disabled', hasExtensions);
+
+            // Disable delete icons and their SVGs
+            mimeTypeField.find('coral-icon[icon="delete"], ._coral-Icon--svg').css({
+                'pointer-events': hasExtensions ? 'none' : 'auto'
+            });
+
+            // Handle MIME type items if extensions exist
+            if (hasExtensions) {
+                // Keep first item, remove others
+                let firstItem = mimeTypeField.find('coral-multifield-item').first();
+
+                // If no first item exists, create one
+                if (firstItem.length === 0) {
+                    // Click the add button to create a new item
+                    mimeTypeField.find('button[coral-multifield-add]').click();
+                    firstItem = mimeTypeField.find('coral-multifield-item').first();
+                }
+
+                mimeTypeField.find('coral-multifield-item:not(:first)').remove();
+                
+                // Set value of first item to */*
+                firstItem.find('input').val('*/*');
+            }
+
+            if(mimeTypeFieldInfoIcon.length > 0) {
+                // show the info icon
+                mimeTypeFieldInfoIcon.css({
+                    'display': hasExtensions ? '' : 'none'
+                });
+            }
+        }
+
+        // Watch for changes in both fields
+        extensionsField.on('change', updateMimeTypeState);
+        mimeTypeField.on('change', updateMimeTypeState);
+
+        // Check initial state without triggering events
+        let hasExtensions = extensionsField.find('coral-multifield-item').length > 0;
+        if (hasExtensions) {
+            mimeTypeField.css({
+                'pointer-events': 'none',
```

Comment:
Html5 allows mimetypes and extensions to be set simultaneously. is there any specific reason for this behavior?